hs_core/views/__init__.py

```diff
@@ -803,6 +803,63 @@ def unshare_resource_with_group(request, shortkey, group_id, *args, **kwargs):
 
     return JsonResponse(ajax_response_data)
 
+
+def undo_share_resource_with_user(request, shortkey, user_id, *args, **kwargs):
+    """this view function is expected to be called by ajax"""
+
+    res, _, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
+    user_to_unshare_with = utils.user_from_id(user_id)
+    ajax_response_data = {'status': 'success'}
+    try:
+        user.uaccess.undo_share_resource_with_user(res, user_to_unshare_with)
+        undo_user_privilege = res.raccess.get_effective_privilege(user_to_unshare_with)
+        if undo_user_privilege == PrivilegeCodes.VIEW:
+            undo_user_privilege = "view"
+        elif undo_user_privilege == PrivilegeCodes.CHANGE:
+            undo_user_privilege = "change"
+        elif undo_user_privilege == PrivilegeCodes.OWNER:
+            undo_user_privilege = "owner"
+        else:
+            undo_user_privilege = 'none'
+        ajax_response_data['undo_user_privilege'] = undo_user_privilege
+
+        if user not in res.raccess.view_users:
+            # user has no explict access to the resource - redirect to resource listing page
+            ajax_response_data['redirect_to'] = '/my-resources/'
```

Comment:
@hyi Yes ```res_raccess.view_users``` includes all users (owners, editors, viewers). This check is necessary since after the authorize() the ```'user``` may have lost access to the resource as a result of unsharing of self from the resource.